contentctl/objects/abstract_security_content_objects/detection_abstract.py

```diff
@@ -338,171 +356,209 @@ def serialize_model(self):
+    def mapDetectionNamesToBaselineObjects(cls, v: list[str], info: ValidationInfo) -> List[Baseline]:
         if len(v) > 0:
-            raise ValueError("Error, baselines are constructed automatically at runtime.  Please do not include this field.")
+            raise ValueError(
+                "Error, baselines are constructed automatically at runtime.  Please do not include this field."
+            )
 
-        
-        name:Union[str,None] = info.data.get("name",None)
+        # TODO (cmcginley): if it's possible in practice for this to return None, we should adjust
+        #   the declared type
+        name: Union[str, dict] = info.data.get("name", None)                                        # type: ignore
         if name is None:
             raise ValueError("Error, cannot get Baselines because the Detection does not have a 'name' defined.")
-         
-        director:DirectorOutputDto = info.context.get("output_dto",None)
-        baselines:List[Baseline] = []
+
+        if info.context is None:
+            raise ValueError("ValidationInfo.context unexpectedly null")
+
+        director: DirectorOutputDto = info.context.get("output_dto", None)
+        baselines: List[Baseline] = []
         for baseline in director.baselines:
-            # This matching is a bit strange, because baseline.tags.detections starts as a list of strings, but 
-            # is eventually updated to a list of Detections as we construct all of the detection objects. 
-            if name in [detection_name for detection_name in baseline.tags.detections if isinstance(detection_name,str)]:
+            # This matching is a bit strange, because baseline.tags.detections starts as a list of strings, but
+            # is eventually updated to a list of Detections as we construct all of the detection objects.
+            if name in [detection_name for detection_name in baseline.tags.detections if isinstance(detection_name, str)]:
                 baselines.append(baseline)
 
         return baselines
 
-    @field_validator('macros',mode="before")
+    @field_validator('macros', mode="before")
     @classmethod
-    def getDetectionMacros(cls, v:list[str], info:ValidationInfo)->list[Macro]:
-        director:DirectorOutputDto = info.context.get("output_dto",None)
-        
-        search:Union[str,dict] = info.data.get("search",None)
-        if not isinstance(search,str):
-            #The search was sigma formatted (or failed other validation and was None), so we will not validate macros in it
+    def getDetectionMacros(cls, v: list[str], info: ValidationInfo) -> list[Macro]:
+        if info.context is None:
+            raise ValueError("ValidationInfo.context unexpectedly null")
+
+        director: DirectorOutputDto = info.context.get("output_dto", None)
+
+        # TODO (cmcginley): what's the typing on this dict?
+        search: Union[str, dict] = info.data.get("search", None)
```

Comment:
if search is a dict, what's the possible typing? is it `dict[str, str]`?

Reply:
The searches that are dicts are sigma searches, so the typing for those should be dict[str,Any]. I will update that typing, and the typing for the search field, to reflect that.